### DISCUSSION TITLE
Let's Share Links to Existing Standards and Specifications

### DISCUSSION THREAD

**Discussion by TrevorFSmith** [omigroup/omigroup#12] [General]
From the kickoff meetings and chats it seems like we ~all agree that we should use existing standards and specs when possible.

Let's use this discussion thread as a reference library of the metaverse-related standards and specs.

We won't use every single standard and spec listed here because we are doing something new. But, if we don't use a standard or spec then we should clearly know what the old work provided and why we don't want it. See [Chesterton's fence](https://en.m.wikipedia.org/wiki/Wikipedia:Chesterton%27s_fence).

**Comment by TrevorFSmith** [comment]
- [Security Assertion Markup Language: SAML](https://wiki.oasis-open.org/security/FrontPage)
- [OAuth 2](https://oauth.net/2/)
- [W3C permissions](https://w3c.github.io/permissions/)
- [W3C trusted user interface](https://www.w3.org/Security/wiki/Trusted_User_Interface)
- [W3C security context UI](https://www.w3.org/TR/wsc-ui/)

- [Khronos glTF](https://github.com/KhronosGroup/glTF/blob/master/README.md)
- [FBX](https://code.blender.org/2013/08/fbx-binary-file-format-specification/)
- [OBJ/MTL](https://en.wikipedia.org/wiki/Wavefront_.obj_file)
- [Khronos COLLADA](https://www.khronos.org/collada/)
- [Pixar USD](https://graphics.pixar.com/usd/docs/index.html#IntroductiontoUSD-FurtherUSDresources)
- [Alembic](http://www.alembic.io/) Material Description Format

- [Web3d standards](https://www.web3d.org/standards): VRML, X3D, and HAnim
- [A-Frame](https://aframe.io/)

- [OWL time](https://www.w3.org/TR/owl-time/), 
- Geodetic datums like [WGS84](https://gisgeography.com/wgs84-world-geodetic-system/) and [GeoPose](https://github.com/opengeospatial/Geopose)

- [IPFS distributed storage](https://github.com/ipfs/specs)
- [WebDAV](https://en.m.wikipedia.org/wiki/WebDAV)

**Comment by TrevorFSmith** [comment]
Hey, @dmitrizagidulin. @robertlong suggested that you might be a person with a good set of links to specifications and standards for distributed identity. Would you be willing to add them here so that the OMI effort has good referenes?

**Comment by humbletim** [comment]
(note: bringing these over from the [XR3ngine Response - Proposed OMI Interfaces](https://github.com/omigroup/OMI/discussions/6) Discussion)

  - [Decentralized Identifiers](https://www.w3.org/TR/did-core/) (DIDs)
  - [Web Bundles](https://web.dev/web-bundles/) (.wbn)
  - [ActivityPub](https://www.w3.org/TR/activitypub/) (decentralized social networking protocol)
      -  [ActivityStream](https://www.w3.org/TR/activitystreams-core/) (vocabulary used by ActivityPub)
          - [JSON-LD](https://www.w3.org/TR/json-ld11/) (JSON-based Serialization for Linked Data)
          - [JSON-LD-API](https://www.w3.org/TR/json-ld-api/) (algorithms for programmatic transformations of JSON-LD)
  - [Credential Handler API](https://w3c-ccg.github.io/credential-handler-api/) (CHAPI)

**Comment by humbletim** [comment]
(intrduced by Vytek in discord [#omi-general](https://discord.com/channels/770382203782692945/830454514682167356/835606582150299718))

- [MMOX](https://tools.ietf.org/id/draft-cscholz-mmox-architecture-00.txt) (2009 IETF draft)
    - Federated/portable identity
    - Distributed Authorization
    - Service Discovery
    - Presence Protocols
    - Licensing/permissions/DRM
    - data formats (e.g. LLSD) 
    - world simulation protocols 
 - [MMOX Use Cases](https://tools.ietf.org/html/draft-jwatte-mmox-use-cases-00)

**Reply by Vytek** [reply]
http://faculty.nps.edu/brutzman/vrtp/  (virtual reality transfer protocol)

**Reply by Vytek** [reply]
https://www.web3d.org/x3d/what-x3d/ (X3D)
https://tecfa.unige.ch/guides/vrml/vrml97/spec/ (VRML)

**Comment by robertlong** [comment]
A list I've had going for a bit:

- Main Topics
  - Authentication
    - Prior Work:
      - OAuth 2.0
        - https://tools.ietf.org/html/rfc6749
      - OpenID Connect
        - https://openid.net/specs/openid-connect-core-1_0.html
      - Web Authentication
        - https://www.w3.org/TR/webauthn/
      - Decentralized Identifiers (DID)
        - https://arxiv.org/pdf/2006.04754.pdf
        - https://github.com/decentralized-identity/did-jwt
        - https://github.com/WebOfTrustInfo/rwot6-santabarbara/blob/master/final-documents/did-auth.pdf
  - Identity
    - Prior Work:
      - OpenID Connect
        - https://openid.net/specs/openid-connect-core-1_0.html#Claims
      - ActivityPub
        - https://www.w3.org/TR/activitypub/#actor-objects
      - Activity Streams
        - https://www.w3.org/TR/activitystreams-core/#actors
        - https://www.w3.org/TR/activitystreams-vocabulary/#actor-types
      - Decentralized Identifiers (DID)
        - https://www.w3.org/TR/did-core/
        - https://www.w3.org/TR/did-use-cases/#messaging
        - https://www.w3.org/TR/did-spec-registries/
        - https://github.com/decentralized-identity/universal-resolver
        - https://github.com/decentralized-identity/ethr-did-resolver
        - https://github.com/decentralized-identity/web-did-resolver
      - Self-Sovereign Identity
        https://github.com/WebOfTrustInfo/rwot6-santabarbara/blob/master/final-documents/a-roadmap-for-ssi.pdf
      - WebID
        - https://dvcs.w3.org/hg/WebID/raw-file/tip/spec/identity-respec.html
      - WebFinger
        - https://tools.ietf.org/html/rfc7033
        - https://diaspora.github.io/diaspora_federation/discovery/webfinger.html
      - hCard
        - http://microformats.org/wiki/hcard
        - https://diaspora.github.io/diaspora_federation/discovery/hcard.html
  - Avatars
    - Prior Work:
      - VRM
        - https://vrm.dev/en/vrm/
        - https://github.com/vrm-c/vrm-specification/tree/master/specification/VRMC_vrm-1.0_draft
  - Social Graph
    - Prior Work:
      - ActivityPub
        - https://www.w3.org/TR/activitypub/#followers
        - https://www.w3.org/TR/activitypub/#following
      - Activity Streams
        - https://www.w3.org/TR/activitystreams-vocabulary/#connections
      - Secure Scuttlebutt
        - https://scuttlebot.io/more/protocols/secure-scuttlebutt.html
      - Diaspora
        - https://diaspora.github.io/diaspora_federation/
  - Messaging / Notifications
    - Prior Work:
      - Matrix
        - https://matrix.org/docs/spec/
        - https://matrix.org/docs/spec/client_server/r0.6.1
        - https://matrix.org/docs/spec/server_server/r0.1.4
        - https://matrix.org/docs/spec/identity_service/r0.3.0
        - https://matrix.org/docs/spec/push_gateway/r0.1.1
      - Push API
        - https://developer.mozilla.org/en-US/docs/Web/API/Push_API
      - Notifications API
        - https://developer.mozilla.org/en-US/docs/Web/API/Notifications_API
  - Inventory
  - Storage
    - IPFS
      - https://docs.ipfs.io/
    - Dat
      - https://www.datprotocol.com/
      - https://docs.dat.foundation/docs/intro
    - Solid
      - https://solidproject.org/about
      - https://solidproject.org/TR/protocol
  - Ownership Verification
    - Prior Work:
      - Verifiable Credentials
        - https://www.w3.org/TR/vc-data-model/
        - https://www.w3.org/TR/vc-use-cases/
      - NFTs

- Implementation
  - API Format
    - JSON-LD
      - https://www.w3.org/TR/json-ld11/
      - https://www.w3.org/TR/json-ld11-api/
    - CBOR-LD
      - https://digitalbazaar.github.io/cbor-ld-spec/
      - https://github.com/digitalbazaar/cborld
  - 3D Content
    - glTF
      - https://github.com/KhronosGroup/glTF/blob/master/specification/2.0/README.md
    - VRM
      - https://vrm.dev/en/vrm/
      - https://github.com/vrm-c/vrm-specification/tree/master/specification/VRMC_vrm-1.0_draft
    - USD
      - https://graphics.pixar.com/usd/docs/index.html

**Comment by mrmetaverse** [comment]
Jin shared this: https://opensea.io/blog/guides/non-fungible-tokens/#ERC721 
Important conversation: https://discord.com/channels/770382203782692945/830454514682167356/839921790025334844

**Comment by dmitrizagidulin** [comment]
Argh, how did I miss the notification for this :) 
@robertlong that is a fantastic starting list! I will add to this shortly.
